app/utils/build-url-with-prefix-map.js

```diff
@@ -0,0 +1,39 @@
+import Ember from 'ember';
+
+export default function buildURLWithPrefixMap(prefixPropertyMapping){
+  return function buildURL(type, id, record) {
+    var url = [],
+        host = Ember.get(this, 'host'),
+        prefix = this.urlPrefix();
```

Comment:
yeah, good point. I lifted this directly from [ember-data's version](https://github.com/emberjs/data/blob/9e9a78111825d4154522cba8ddbbe679c56c9e5c/packages/ember-data/lib/adapters/rest_adapter.js#L532-L550), which looked like that. I'd prefer to minimize the diffs between this and that.

I'm gonna add in a link to the ember-data function, though.